src/core/expr/fexpr_alias.cc

```diff
@@ -0,0 +1,122 @@
+//------------------------------------------------------------------------------
+// Copyright 2019-2022 H2O.ai
+//
+// Permission is hereby granted, free of charge, to any person obtaining a
+// copy of this software and associated documentation files (the "Software"),
+// to deal in the Software without restriction, including without limitation
+// the rights to use, copy, modify, merge, publish, distribute, sublicense,
+// and/or sell copies of the Software, and to permit persons to whom the
+// Software is furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+// IN THE SOFTWARE.
+//------------------------------------------------------------------------------
+#include "column/const.h"
+#include "documentation.h"
+#include "expr/fexpr_func.h"
+#include "expr/eval_context.h"
+#include "expr/workframe.h"
+#include "python/xargs.h"
+#include "stype.h"
+
+
+namespace dt {
+namespace expr {
+
+class FExpr_Alias : public FExpr_Func {
+  private:
+    ptrExpr arg_;
+    strvec names_;
+
+  public:
+    FExpr_Alias(ptrExpr&& arg, strvec&& names)
+      : arg_(std::move(arg)), 
+        names_(std::move(names))
+        {}
+
+    std::string repr() const override {
+      std::string out = "alias";
+      out += '(';
+      out += arg_->repr();
+      out += ", names=";
+      out += "[";
+      for (auto name : names_) {
+        out += name;
+        out += ",";
+      }
+        out += "]";
+      out += ')';
+      return out;
+    }
+
+
+    Workframe evaluate_n(EvalContext& ctx)  const override{
+      Workframe wf = arg_->evaluate_n(ctx);
+      if (wf.ncols() != names_.size()) {
+        throw ValueError() << "The number of arguments("<<names_.size()<<") "
+            "passed to `names` does not match the number of specified columns"
+            "("<<wf.ncols()<<").";
+      }
+      auto gmode = wf.get_grouping_mode();
+      Workframe outputs(ctx);
+
+      for (size_t i = 0; i < wf.ncols(); ++i) {
+        Workframe base_frame(ctx);
+        base_frame.add_column(wf.retrieve_column(i),
+                              std::string(),
+                              gmode);
+        base_frame.rename(names_[i]);
```

Comment:
My feeling is that `.rename()` should be able to rename a set of columns to different names. Right now this method only accepts one name for all columns and in some cases it actually adds a prefix...